src/core/attributeformmodelbase.cpp

```diff
@@ -374,9 +374,11 @@ void AttributeFormModelBase::flatten( QgsAttributeEditorContainer *container, QS
 
         // create constraint description
         QStringList descriptions;
-        if ( !field.constraints().constraintDescription().isEmpty() )
+        if ( field.constraints().constraints() & QgsFieldConstraints::ConstraintExpression )
         {
-          descriptions << field.constraints().constraintDescription();
+          descriptions << ( !field.constraints().constraintDescription().isEmpty()
+                            ? field.constraints().constraintDescription()
+                            : tr( "Expression constraint" ) );
```

Comment:
"Constraint check failed"?

Reply:
@m-kuhn , we can't phrase it that way. The description here is only used to describe which constraint(s) is/are one, not whether the constraint is met or not. For e.g., we don't say "unique value failed", we just say "unique value".

In the future, it might be worth updating the description to only state which constraint(s) is/are failing. That'd require code upgrade in QGIS and QField.

But for now, esp. for 1.9, that's the best we can do, which is already IMHO an improvement.

Reply:
It is definitely better, merge ahead.